### ISSUE TITLE
S3928 false positive when analyzing indexers

### ISSUE THREAD

**Issue by ondrej2286** [SonarSource/sonar-dotnet#2469]
### Description

Warning S3928 given when analyzing a C# indexer.

### Repro steps

Consider the following code in a C# class:
```csharp
using System;
namespace S3928.Failure
{
    public class SampleClass
    {
        public bool this[string action]
            => Get(action ?? throw new ArgumentNullException(nameof(action)));

        private bool Get(string action) => action == this.ToString();
    }
}
```
The rule fails on line 7 with: 
`"Warning S3928 The parameter name 'action' is not declared in the argument list."`

### Expected behavior

No warning is given.

### Actual behavior

Warning S3928 given.

### Known workarounds

Turn this rule off.

### Related information

* SonarLint for Visual Studio 2019 version 4.10.0.9867
* Visual Studio 2019 version 16.1.4

**Comment by christophe-zurn-sonarsource** [comment]
I confirm this is indeed a FP on the analyzer side.

Method [CheckArgumentException#GetMethodArgumentNames](https://github.com/SonarSource/sonar-dotnet/blob/0ca63df263f33e47296f6e95f38878987b9f9fa7/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs#L123) should be able to correctly resolve the argument names when an ancestor of type `IndexerDeclarationSyntax` is found.

```
namespace Tests.Diagnostics
{
    class Program
    {
        public bool this[string a] => throw new ArgumentNullException(nameof(a)); // FP
    }
}
```
